hs_discover/views.py

```diff
@@ -223,31 +216,26 @@ def get(self, request, *args, **kwargs):
                 author = str(result.author)
                 if authors == 'None':
                     authors = author  # author would override creator in
-            else:
-                if result.organization:
-                    if isinstance(result.organization, list):
-                        author = str(result.organization[0])
-                    else:
-                        author = str(result.organization)
 
-                    author = author.replace('"', '')
-                    author = author.replace('[', '')
-                    author = author.replace(']', '').strip()
+            elif result.organization:
+                if isinstance(result.organization, list):
+                    author = str(result.organization[0])
+                else:
+                    author = str(result.organization)
+
+                author = author.replace('"', '')
+                author = author.replace('[', '')
+                author = author.replace(']', '').strip()
```

Comment:
These 3 lines are odd.  What's the purpose?  Is author a list cast to a string or something?

Reply:
I don't get them either. It seems that organization might be a list. I need to investigate.

Reply:
At a deeper level, if we are going to treat organization as author, they should also be listed in the author field/facet. We shouldn't cobble them in here. This hits Django, which slows down the whole process.

Reply:
I checked. Organization is NEVER a list and no organization in all of hydroshare contains the characters [. ], ". This is a no-op.

Reply:
Oops. This is a SOLR query result. In SOLR, organizations is ALWAYS a list. The question is what we do with something that has no author and has a list of organizations. What Mark does is to concatenate them. I'm not sure that this is the best idea.  I think returning the first one is reasonable.

Reply:
At this point in the code: resource is a SOLR query response. This has elements returned by SOLR and elements that are resolved by querying Django. In the normal Django model there is no resource.organization. It's resource.metadata.organization.

Reply:
This is more confused than I thought. Turns out that SOLR automatically inserts the organization into the author field if there is no author. So, unless SOLR breaks big-time, this should never even be executed.

Reply:
Out of time for today. "Sorted out" sorting and SOLR references. Everything seems to still work.